File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n=len(arr)
        suffix_max=-1
        for i in range(n-1,-1,-1):
            cur=arr[i]
            arr[i]=suffix_max
            suffix_max=max(suffix_max,cur)
        return arr

File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int n=arr.length;
        int suffix_max=-1;
        for(int i=n-1;i>=0;i--)
        {
            int cur=arr[i];
            arr[i]=suffix_max;
            suffix_max=Math.max(suffix_max,cur);
        }
        return arr;
    }
}